### PR TITLE
feat(configs): add community SO101 dataset mixture config

### PR DESCRIPTION
**Pull request by WilliamYue37**
## What this does
Adds `configs/examples/refs/so101_community_mixture.json`, a ref fragment listing 3,240 open-source LeRobot SO101 datasets from the HF Hub, curated for training co-training mixtures: 343,435 episodes / 170.6M frames across single-arm and bimanual entries, each with a per-instance `data_features_name_mapping` resolved against that repo's actual camera/state/action column names.

**Discovery.** Started from name/tag search (`so101`, `so-101`, `so_101`, `so-arm101`, etc. — 8,676 + 148 candidates across two rounds), then closed the gap with an exhaustive pass: HF's `search`/`filter` params do token-level full-text matching, not raw substring matching, so an id like `biso101_put_red_block_into_bin` (one token) never matches a `so101` search even though it contains that literal substring — same story for the `soarm101` (no separator) naming variant several uploaders use. To guarantee nothing was missed, I enumerated all ~959k public HF datasets and regex-matched every repo id client-side (`so[-_]?101|so[-_]?arm[-_]?101`), which surfaced 148 repos the text-search rounds had missed; 45 of those survived the same filters below.

**Curation pipeline** (from ~8,824 total candidates across all rounds):
- **Automated filters**: no `test`/`temp` in the name, >5 episodes, every episode >1s, `meta/info.json` codebase_version readable by OpenTau's loader (v2.0-v3.0), >=1 video feature, fps in [10, 60], >=1 real (non-placeholder) task string, non-degenerate action stats, no `eval_`-prefixed policy-rollout recordings.
- **Programmatic dedupe**: exact re-upload fingerprints (identical episode/frame/fps/camera/task signature) and exact-arithmetic "merge of siblings" detection (e.g. a `_merged` repo whose episode/frame counts are the precise sum of other same-author repos already in the set) — 58 caught this way.
- **Loader verification**: every surviving repo's metadata was actually loaded through OpenTau's `LeRobotDatasetMetadata`, not just inspected via raw HTTP — this caught 93 repos that pass every metadata heuristic but fail to load (stale Hub version tags resolving to an incompatible pre-v2.0 format, episode-shard schema drift, etc).
- **LLM review pass with adversarial verification**: batches of records and the largest datasets' cards were reviewed for signal regexes can't catch, then every proposed exclusion was independently re-verified against the actual per-episode stats/video/dataset card before being dropped (27 confirmed, 21 rejected as false positives). Caught things like reward-classifier *negative/failure* demonstrations, placeholder task labels (`"1st Training"`, `"my_first_task"`), smoke-test/calibration recordings, and a byte-level LFS-hash-verified duplicate upload.

Weights are intentionally left unset — `DatasetMixtureConfig.weights=None` makes OpenTau infer per-dataset weight from each dataset's actual loaded length (`datasets/factory.py::_resolve_weights`), i.e. proportional-to-size computed at build time rather than a precomputed, potentially stale, frame count.

Also:
- Allowlists a handful of literal (mis-spelled) upstream camera column names — e.g. `observation.images.righ`, `observation.images.sholder` — in the `typos` config, since these are the *actual* column names in those specific HF datasets and must be preserved verbatim for the mapping to resolve.
- Excludes this one large generated file from the `check-added-large-files` guard (it's an intentionally large, reviewed list of 3,240 datasets, not an accidentally committed artifact).

## How it was tested
- The fragment decodes cleanly via `draccus.decode(DatasetMixtureConfig, ...)` with 3,240 unique `repo_id`s.
- Every entry was independently confirmed to load via `LeRobotDatasetMetadata` (the same class used at real training time), with its mapped camera/state/action columns checked against the dataset's actual `features`.
- `pre-commit run --all-files` passes.
- `pytest -m "not gpu and not network" tests/configs tests/datasets -q` → 725 passed, 7 skipped (env-marker skips unrelated to this change).

## How to checkout & try? (for the reviewer)
```bash
python -c "
import json, draccus
from opentau.configs.default import DatasetMixtureConfig
cfg = json.load(open('configs/examples/refs/so101_community_mixture.json'))
mixture = draccus.decode(DatasetMixtureConfig, cfg)
print(len(mixture.datasets), 'datasets')
"
```
To use in a training config, reference it the same way `train_with_ref.json` references `droid_mixture.json`:
```json
{"dataset_mixture": {"$ref": "so101_community_mixture.json"}}
```

## Checklist
- [x] I have added Google-style docstrings to important functions and ensured function parameters are typed.
- [ ] My PR includes policy-related changes.
  - [ ] If the above is checked: I have run the GPU pytests (pytest -m "gpu") and regression tests.

### Note: Before submitting this PR, please read the [contributor guideline](https://github.com/TensorAuto/OpenTau/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: configs/examples/refs/so101_community_mixture.json [hunk 1]
```diff
@@ -0,0 +1 @@
...}},{"repo_id":"SurajChess/so101_dataset3","data_features_name_mapping":{"camera0":"observation.images.front","camera1":"observation.images.wrist","state":"observation.state","actions":"action","prompt":"task"}},{"repo_id":"SurajChess/so101_dataset_marker","data_features_name_mapping":{"camera0":"observation.images.front","camera1":"observation.images.wrist","state":"observation.state","actions":"action","prompt":"task"}},{"repo_id":"SurajChess/so101_dataset_marker_tape","data_features_name_mapping":{"camera0":"observation.images.front","camera1":"observation.images.wrist","state":"observation.state","actions":"action","prompt":"task"}},{"repo_id":"SurajChess/so101_dataset_marker_tape_1","data_features_name_mapping":{"camera0":"observation.images.front","camera1":"observation.images.wrist","state":"observation.state","actions":"action","prompt":"task"}},{"repo_id":"SurajChess/so101_dataset_tape","data_features_name_mapping":{"camera0":"observation.images.front","camera1":"observation.images.wrist","state":"observation.state","actions":"action","prompt":"task"}},{"repo_id":"SurajChess/so101_dataset_tape1","data_features_name_mapping":{"camera0":"observation.images.front","camera1":"o
```
[the remark this answers is not in the document]
Comment (WilliamYue37) [reply configs/examples/refs/so101_community_mixture.json]: @claude fix 1 only

**Comment by WilliamYue37** [comment]
## Mock training run / smoke test results

Ran a real (not metadata-only) construction smoke test: built actual `LeRobotDataset` objects for a stratified 186-dataset sample (covering every camera-count bucket 1-5, bimanual entries, both discovery rounds, and the highest-frame-count datasets), each restricted to `episodes=[0]` to keep downloads small, then combined the successful ones into a real `WeightedDatasetMixture` and pulled batches from its `DataLoader`.

**Result:** 181/186 (97%) constructed cleanly, and the mixture produced correctly-padded batches end to end (`camera0-4`, `state`→32-dim, `actions`→`(10, 32)`, plus all the metadata fields the training loop consumes).

**5 confirmed broken, now dropped** (10ac1d7) — these pass every metadata-only check (which never touches actual parquet/video data) but fail real construction:
- `Gayagaya/SO101_dataset` — malformed parquet schema, missing `episode_index`
- `HBOrtiz/so101_eval3_broad`, `HBOrtiz/so101_eval3_cotrain` — malformed `meta/stats.json` (`count` stat shaped `(3,)` instead of `(1,)`)
- `SkywalkerLi/so101_phase_split` — 404 on its episode-0 video file
- `arunabh-ramesh/so101_lift_real` — repo deleted from the Hub entirely since the original scan

**2 more that only fail under the smoke test's own `episodes=[0]` restriction** — confirmed as a testing-methodology artifact, not a real config problem, so left in the mixture: `LeRobotDataset` only recomputes `meta.stats` from a per-episode subset when `episodes` is explicitly set (`lerobot_dataset.py:1734`); `gpudad/so101_pick_cube` and `gpudad/so101_pick_cube_chunked`'s episode 0 happens to have a stats gap that the full-dataset aggregate doesn't. Since this config never sets `episodes` on any entry, real training never hits that code path.

Final count: **3,235 datasets**.

**Review by shuheng-liu — APPROVED** [review]
